build/Makefile

```diff
@@ -0,0 +1,169 @@
+SHELL:=/usr/bin/env bash
+SRC_DIR:=../src
+SRC_FILES:=$(shell find $(SRC_DIR) -name '*.xml' -o -name '*.json')
+SRC_XML_PROFILES:=$(shell find $(SRC_DIR) -name '*profile.xml')
+SRC_READMES:=$(shell find $(SRC_DIR) -iname 'README.md')
+GEN_CONTENT_DIR:=generated
+GEN_READMES:=$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_READMES))
+GEN_XML_FILES:=$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_FILES))
+GEN_XML_PROFILES:=$(subst _profile.xml,-resolved-profile_catalog.xml,$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_XML_PROFILES)))
+GEN_JSON_FILES:=$(subst xml,json,$(GEN_XML_FILES))
+GEN_MIN_JSON_FILES:=$(subst .json,-min.json,$(subst xml,json,$(GEN_XML_FILES)))
+GEN_YAML_FILES:=$(subst xml,yaml,$(GEN_XML_FILES))
+CURL_INSTALL_OPTS:=--silent --location
+JQ_BIN:=jq-linux-amd64
+JQ_PATH:=$(shell pwd)/jq
+JQ_INSTALL_VERSION:=jq-1.7
+JQ_DOWNLOAD_URL:=https://github.com/jqlang/jq/releases/download/$(JQ_INSTALL_VERSION)/$(JQ_BIN)
+JQ_INSTALL_COMMAND=curl $(CURL_INSTALL_OPTS) -o $(JQ_PATH) $(JQ_DOWNLOAD_URL) && chmod +x $(JQ_PATH)
+XMLLINT_PATH=/usr/bin/xmllint
+XMLLINT_INSTALL_COMMAND=sudo rm -rf /var/lib/apt/lists/* && sudo apt-get update && sudo apt-get install libxml2-utils
+YQ_BIN:=yq_linux_amd64
+YQ_PATH:=$(shell pwd)/yq
+YQ_INSTALL_VERSION:=v4.35.2
+YQ_DOWNLOAD_URL:=https://github.com/mikefarah/yq/releases/download/$(YQ_INSTALL_VERSION)/$(YQ_BIN)
+YQ_INSTALL_COMMAND:=curl $(CURL_INSTALL_OPTS) -o $(YQ_PATH) $(YQ_DOWNLOAD_URL) && chmod +x $(YQ_PATH)
+NPM_PREFIX_DIR:=oscal/build
+NPM_PKGS_DIR:=node_modules
+XSLT_RUNNER:=oscal/build/xslt-runner.sh
+PROFILE_RESOLVER_RUNNER:=oscal/src/utils/resolver-pipeline/oscal-profile-resolve.sh
+PROFILE_RESOLVER_ARGS:="uuid-method='random-xslt'"
+XML_JSON_CONVERTER_XSLT:=oscal/build/generated/oscal_complete_xml-to-json-converter.xsl
+OSCAL_COMPLETE_XML_SCHEMA:=oscal/build/generated/oscal_complete_schema.xsd
+OSCAL_COMPLETE_JSON_SCHEMA:=oscal/build/generated/oscal_complete_schema.json
+
+.PHONY: help
+# Run "make" or "make help" to get a list of user targets
+# Adapted from https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
+help: ## Show this help message
+	@grep -E '^[a-zA-Z_-]+:.*?##.*$$' $(MAKEFILE_LIST) | awk 'BEGIN { \
+	 FS = ":.*?## "; \
+	 printf "\033[1m%-30s\033[0m %s\n", "TARGET", "DESCRIPTION" \
+	} \
+	{ printf "\033[32m%-30s\033[0m %s\n", $$1, $$2 }'
+
+.PHONY: all
+all: \
+	dependencies             \
+	build                    \
+	copy-readmes             \
+	copy-xml-content         \
+	resolve-xml-profiles     \
+	convert-min-json-content \
+	reformat-json-content    \
+	convert-yaml-content     \
+	validate-xml-content     \
+	validate-json-content    \
+	validate-yaml-content ## Run all steps for content preparation
```

Comment:
NB: A limitation of the `help` target is that it assumes targets are all on one line. I like the readability afforded by your multiline target, however it will not be printed by `make help` in its current form :(. I leave this trade-off to your judgement.

Reply:
Ohhh, I hadn't even noticed this. 😢 That can be adjusted, I just skimmed the sub-targets. I shake my fist at the `make`-makers.

Reply:
Unfortunately it is me you should be shaking your fist at me, the `help` target just does not expect multi-line target dependencies. It's possible you can adjust the `grep` and `awk` commands, but that sounds like it could be more trouble than it's worth.

Reply:
I'd rather not get too deep into that, I can remove that change in upcoming commits and rebasing (I think this had a dedicated cleanup commit, I will update on the change).

Reply:
Done in https://github.com/usnistgov/oscal-content/pull/204/commits/caec68c96437997e1253467a5d8b499feb557d07.